### PR TITLE
Item repo2

### PR DESCRIPTION
**Pull request by ochar721**

### CODE REVIEW

File: lib/item_repository.rb [hunk 1]
```diff
@@ -16,9 +18,72 @@ def generate_items(filename)
...
+    id = @items[-1].id.to_i
+    id += 1
+    id = id.to_s
+    attributes[:id] = id
+    string_unit_price = attributes[:unit_price].to_f
+    attributes[:unit_price] = string_unit_price
+    item = Item.new(attributes)
+    @items.push(item)
+  end
+
```
Comment (Pragmaticpraxis37) [inline lib/item_repository.rb:76]: It looks like we could break out a helper method on lines 63 - 66.

File: lib/item_repository.rb [hunk 2]
```diff
@@ -16,9 +18,72 @@ def generate_items(filename)
...
+      item.merchant_id == merchant_id.to_s
+    end
+  end
+
+  def create(attributes)
+    id = @items[-1].id.to_i
+    id += 1
+    id = id.to_s
+    attributes[:id] = id
+    string_unit_price = attributes[:unit_price].to_f
```
Comment (Pragmaticpraxis37) [inline lib/item_repository.rb:71]: We end up converting the prices to integers and back to string, maybe we can create a helper method.

File: lib/item_repository.rb [hunk 3]
```diff
@@ -16,9 +18,72 @@ def generate_items(filename)
...
+  end
+
+  def find_all_with_description(description)
+    @items.find_all do |item|
+      item.description.downcase == description.downcase
+    end
+  end
+
+  def find_all_by_price(price)
+    price_fix = "#{price}00"
```
Comment (Pragmaticpraxis37) [inline lib/item_repository.rb:44]: We might need to create a helper method for this one.

**Review by Pragmaticpraxis37 — COMMENTED** [review]
Looks good overall, I think we can shorten up some methods by using helper methods.

File: lib/item_repository.rb (hunk 1 above)
Comment (ochar721) [reply lib/item_repository.rb:76]: okay, that makes sense.  we can change it into a helper method

File: lib/item_repository.rb (hunk 2 above)
Comment (ochar721) [reply lib/item_repository.rb:71]: Can you explain that a little more in a paired setting

File: lib/item_repository.rb (hunk 3 above)
Comment (ochar721) [reply lib/item_repository.rb:44]: Okay, we can take a look

File: lib/item_repository.rb (hunk 1 above)
Comment (Pragmaticpraxis37) [reply lib/item_repository.rb:76]: Sounds good.

File: lib/item_repository.rb (hunk 3 above)
Comment (Pragmaticpraxis37) [reply lib/item_repository.rb:44]: Excellent.

File: lib/item_repository.rb (hunk 2 above)
Comment (Pragmaticpraxis37) [reply lib/item_repository.rb:71]: Sure, a phone conversation about this might be easier.

**Comment by jequeza** [comment]
Fixed conflicts @everyone!!

**Pull request merged**